command/kv_list.go

```diff
@@ -75,25 +93,56 @@ func (c *KVListCommand) Run(args []string) int {
 		return 2
 	}
 
-	// Sanitize path
-	path := sanitizePath(args[0])
-	mountPath, v2, err := isKVv2(path, client)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 2
-	}
+	// If true, we're working with "-mount=secret foo" syntax.
```

Comment:
logic is the same as in kv put
https://github.com/hashicorp/vault/blob/d35be2d0de3d1c036248570c538c2051c4c1dc57/command/kv_put.go#L144C2-L176